main.go

```diff
@@ -326,10 +347,23 @@ func main() {
 	log.Printf("Took: " + time.Since(startTime).String())
 }
 
-func isSameStack(archiveStackID string, currentStackID string) bool {
+func isSameStack(archiveStackInfo archiveInfo, currentStackInfo archiveInfo) bool {
```

Comment:
Some utility functions are defined at the top of the file, others at the bottom.

I propose that we make `main`  the first function (after struct functions).